### PR TITLE
Strict Kustomize Patching

### PR DESCRIPTION
**Pull request by davidz627**
Make it so that `base != stable`, this makes versioning crystal clear

/assign @msau42

### CODE REVIEW

File: deploy/kubernetes/overlays/dev/controller_images.yaml [hunk 1]
```diff
@@ -6,6 +6,12 @@ spec:
   template:
     spec:
       containers:
+        - name: csi-provisioner
+          image: gcr.io/gke-release-staging/csi-provisioner:v0.4.1-gke.0
```
Comment (msau42) [inline deploy/kubernetes/overlays/dev/controller_images.yaml]: gke-release-staging is not publicly accessible, so we need to use gke-release or quay here.  Same with stable and prow-head-template (which is used in oss pull jobs)

Comment (davidz627) [reply deploy/kubernetes/overlays/dev/controller_images.yaml]: ah my bad, botched copy-paste job

**Comment by davidz627** [comment]
@msau42 should be fixed

File: deploy/kubernetes/overlays/dev/controller_images.yaml [hunk 2]
```diff
@@ -6,6 +6,12 @@ spec:
   template:
     spec:
       containers:
+        - name: csi-provisioner
```
Comment (msau42) [inline deploy/kubernetes/overlays/dev/controller_images.yaml]: Separately we should also consider if we want a job that tests latest sidecars too

File: deploy/kubernetes/overlays/dev/controller_images.yaml [hunk 3]
```diff
@@ -6,6 +6,12 @@ spec:
...
+          image: gcr.io/gke-release/csi-provisioner:v0.4.1-gke.0
+        - name: csi-attacher
+          image: gcr.io/gke-release/csi-attacher:v0.4.1-gke.0
+        - name: gce-pd-driver
```
Comment (msau42) [inline deploy/kubernetes/overlays/dev/controller_images.yaml]: there's already a gce-pd-driver entry below

File: deploy/kubernetes/overlays/prow-head-template/node_images.yaml [hunk 4]
```diff
@@ -8,3 +8,5 @@ spec:
       containers:
         - name: gce-pd-driver
           image: PROW_GCEPD_IMAGE:latest
+        - name: gce-pd-driver
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-head-template/node_images.yaml]: remove?

Comment (davidz627) [reply deploy/kubernetes/overlays/prow-head-template/node_images.yaml]: should be driver registrar

File: deploy/kubernetes/overlays/dev/controller_images.yaml (hunk 3 above)
Comment (davidz627) [reply deploy/kubernetes/overlays/dev/controller_images.yaml]: oops

File: deploy/kubernetes/overlays/dev/controller_images.yaml (hunk 2 above)
Comment (davidz627) [reply deploy/kubernetes/overlays/dev/controller_images.yaml]: prow-gke-head does that

**Comment by davidz627** [comment]
@msau42 fixed

**Pull request merged**